lib/filesystem.js

```diff
@@ -73,22 +73,36 @@ class Filesystem {
       return callback()
     }
 
-    const tr = options.transform && options.transform(p)
-    if (tr) {
-      return tmp.file(function (err, path) {
-        if (err) { return handler() }
-        const out = fs.createWriteStream(path)
-        const stream = fs.createReadStream(p)
-
-        stream.pipe(tr).pipe(out)
-        return tr.on('end', function () {
-          file.transformed = {
-            path,
-            stat: fs.lstatSync(path)
-          }
+    if (options.transform instanceof Function) {
+      options.transform = [options.transform]
+    }
+
+    if (options.transform instanceof Array && options.transform.length) {
+      options.transform = options.transform.filter(() => true)
+
+      const doTransform = (path, transformers, idx) => {
+        idx = idx || 0
+        const transform = transformers.hasOwnProperty(idx) ? transformers[idx](p) : false
```

Comment:
hasOwnProperty is in case the Array prototype has been messed with. Again, perhaps overly defensive. We check if there's a transform function to use at the start of doTransform, but you're right; we could just as easily check if there's another transform function at the end and only recurse if we need to